internal/pkg/pkgfile/build.go

```diff
@@ -145,8 +148,40 @@ func Build(ctx context.Context, c client.Client, options *environment.Options) (
 				return err
 			}
 
+			var cacheImports []client.CacheOptionsEntry
+
+			// new API
+			if cacheImportsStr := opts[keyCacheImports]; cacheImportsStr != "" {
```

Comment:
we could actually pass in json and drop old api support at all if needed

Reply:
I'm not sure if buildx supports new API fully, this "new" API is 5 years old now